Availability bar, progress bar, button group improvements

Chocobo1 opened the pull request:
Tooltip for progress bar:
![1](https://cloud.githubusercontent.com/assets/9395168/8403719/896cc494-1e79-11e5-8ffe-ab4117ab6906.png)

Tooltip for availability bar:
![2](https://cloud.githubusercontent.com/assets/9395168/8403720/8bb28cac-1e79-11e5-8238-492320ef3cf1.png)

-----

UPDATE:
`Use theme color for background in PropertiesWidget` & `Replace horizontal line with border` changes this:

Before:
![2](https://cloud.githubusercontent.com/assets/9395168/8637503/85acb902-28c6-11e5-937c-8d732f8bb758.png)

After:
![new](https://cloud.githubusercontent.com/assets/9395168/8637502/840f9a6a-28c6-11e5-9f42-ecb382ba93f5.png)

ngosang commented:
> Green: Pieces we have a partial of it

Improve this translation. Maybe "Green: Pieces being downloaded"

There is an empty space when the Availability bar is hidden.
![](http://i.imgur.com/W3jvJ30.png)

Progress and Availability bars are higher now. My screen has a normal DPI and prefer to occupy less space, the same as before.

I can't test how this is displayed in a monitor with high DPI.

Chocobo1 commented:
>Improve this translation. Maybe "Green: Pieces being downloaded"

I am not sure, there is also a pause state which your suggestion would be wrong.

> There is an empty space when the Availability bar is hidden.

I will look into that.

>Progress and Availability bars are higher now

If you can look at here: https://github.com/Chocobo1/qBittorrent/commit/64608ff934b6b5a99b344b298ed95987c5cf7a9e#diff-0c38e7586d610b43e1f65b4fd3ed7ce0R110
and give me a better formula that would be great, instead of me guessing what you prefer.

>I can't test how this is displayed in a monitor with high DPI.

In cinnamon DE, there is a setting to force high DPI display, not sure about others.

ngosang commented:
> give me a better formula that would be great, instead of me guessing what you prefer.

This displays like before but maybe it's wrong for HiDPI monitors.
```
int barHeight = fMetrics.height() * 1.1;
```

Chocobo1 commented:
> int barHeight = fMetrics.height();

That is too small for me, probably we use different font size/family, so there is different result...
I started wondering if following font height was a good idea.

Anyway, I changed it to:
`int barHeight = fMetrics.height() * 5 / 4;`
See if it's ok on your system.

I also added another code path for qt5 which should be more correct but it's not available on qt4.

ngosang commented:
```
int barHeight = fMetrics.height() * 5 / 4
```
Still big for me but wait for other opinions.

Chocobo1 commented:
Updated.

ngosang commented:
Looks good but with this changes the lower panel height increased by 20px. This is how it looks in KDE 5 Breeze theme.
![](http://i.imgur.com/2QNakxt.png)
Maybe you can reduce the spaces marked with red in the screenshot.

Chocobo1 commented:
> Maybe you can reduce the spaces marked with red in the screenshot.

Done, I shrink the top & bottom space of 3 groups (bar, transfer, information), can retest now.

ngosang commented:
Better now.

sledgehammer999 commented:
Can you rebase? Plus drop the "Legend:" strings. The other strings might need improvement. I'll tell you in my review after the rebase.

Chocobo1 commented:
> drop the "Legend:" strings. 

LIke this?
``` c++
setToolTip(QString("%1\n%2\n%3").arg(tr("White: Pieces we don't have")).arg(tr("Green: Pieces we have a partial of it")).arg(tr("Blue: Pieces we already have")));
```
or delete the whole line?

sledgehammer999 commented:
Only the 2 strings that read "Legend:". Not the ones that explain the colors.

Chocobo1 commented:
>Can you rebase? 
>Only the 2 strings that read "Legend:". Not the ones that explain the colors.

Done.

sledgehammer999 commented:
The code for me is ok. The interface is also ok on Windows 7, XP and debian XFCE. (tested only qt4).

About the tooltips how does the following sound?
**Progress:**
White: Missing pieces
Green: Partial pieces
Blue: Completed pieces

**Availability:**
White: Unavailable pieces
Blue: Available pieces

Of course, we could also drop the word "pieces". Do you guys have any input?

Chocobo1 commented:
>Of course, we could also drop the word "pieces". 

IMO "pieces" should stay, so users will know what kind of unit it is showing.

PR updated.

sledgehammer999 commented:
Thanks for this.
